rflinkproxy/__main__.py

```diff
@@ -0,0 +1,315 @@
+            self.loop.create_task(self.connect())
+
+    async def connect(self):
+        """Set up connection and hook it into HA for reconnect/shutdown."""
+        import serial
+
+        log.info('Initiating Rflink connection')
+
+        # Rflink create_rflink_connection decides based on the value of host
+        # (string or None) if serial or tcp mode should be used
+
+        # Setup protocol
+        protocol = partial(
+            ProxyProtocol,
+            disconnect_callback=self.reconnect,
+            raw_callback=self.raw_callback,
+            loop=self.loop,
+        )
+
+        # Initiate serial/tcp connection to Rflink gateway
+        if self.host:
+            connection = self.loop.create_connection(protocol, self.host, self.port)
+        else:
+            connection = create_serial_connection(self.loop, protocol, self.port, self.baud)
+
+        try:
+            with async_timeout.timeout(CONNECTION_TIMEOUT,
+                                       loop=self.loop):
+                self.transport, self.protocol = await connection
+
+        except (serial.serialutil.SerialException, ConnectionRefusedError,
+                TimeoutError, OSError, asyncio.TimeoutError) as exc:
+            reconnect_interval = DEFAULT_RECONNECT_INTERVAL
+            log.error(
+                "Error connecting to Rflink, reconnecting in %s",
+                reconnect_interval)
+
+            self.loop.call_later(reconnect_interval, self.reconnect, exc)
+            return
+
+        log.info('Connected to Rflink')
+
+
+def main(argv=sys.argv[1:], loop=None):
+    """Parse argument and setup main program loop."""
+    args = docopt(__doc__, argv=argv,
+                  version=pkg_resources.require('rflink')[0].version)
+
+    level = logging.ERROR
+    if args['-v']:
+        level = logging.INFO
+    if args['-v'] == 2:
+        level = logging.DEBUG
+    logging.basicConfig(level=level)
+
+    if not loop:
+        loop = asyncio.get_event_loop()
+
+    host = args['--host']
+    port = args['--port']
+    baud = args['--baud']
+    listenport = args['--listenport']
+
+    proxy = RFLinkProxy(port=port, host=host, baud=baud, loop=loop)
+
+    server_coro = asyncio.start_server(
+        proxy.client_connected_callback,
+        host="",
+        port=listenport,
+        loop=loop,
+    )
+
+    server = loop.run_until_complete(server_coro)
+    addr = server.sockets[0].getsockname()
+    log.info(f'Serving on {addr}')
+
+    conn_coro = proxy.connect()
+    loop.run_until_complete(conn_coro)
+
+    proxy.closing = False
+    try:
+        loop.run_forever()
+    except KeyboardInterrupt:
+        proxy.closing = True
+
+        # cleanup server
+        server.close()
+        loop.run_until_complete(server.wait_closed())
+
+        # cleanup server connections
+        writers = [i[1] for i in list(clients)]
+        for writer in writers:
+            writer.close()
```

Comment:
Is there any reason to wait for the writer to close? Otherwise i would suggest just dropping the python3.7 specific part. 

Or something like this (but not waiting for close on prior pythons could cause issues then?):
```suggestion
            if sys.version_info < (3, 7):
              writer.close()
            else:
              loop.run_until_complete(writer.wait_closed())
```